src/ui/viewmodels/RichPresenceMonitorViewModel.cpp

```diff
@@ -0,0 +1,37 @@
+#include "RichPresenceMonitorViewModel.hh"
+
+#include "RA_GameData.h"
+#include "RA_RichPresence.h"
+
+namespace ra {
+namespace ui {
+namespace viewmodels {
+
+const StringModelProperty RichPresenceMonitorViewModel::DisplayStringProperty("RichPresenceMonitorViewModel", "Message", L"No Rich Presence defined.");
+
+RichPresenceMonitorViewModel::RichPresenceMonitorViewModel() noexcept
+{
+    SetWindowTitle(L"Rich Presence Monitor");
+}
+
+void RichPresenceMonitorViewModel::UpdateDisplayString()
+{
+    auto nGameId = g_pCurrentGameData->GetGameID();
+    if (nGameId == 0)
+    {
+        SetDisplayString(L"No game loaded.");
+    }
+    else if (!g_RichPresenceInterpreter.Enabled())
+    {
+        SetDisplayString(DisplayStringProperty.GetDefaultValue());
+    }
+    else
+    {
+        std::wstring sDisplayString = ra::Widen(g_RichPresenceInterpreter.GetRichPresenceString());
+        SetDisplayString(sDisplayString);
+    }
+}
```

Comment:
`UpdateDisplayString` is called every minute to update the view model message. When it changes, the binding is notified, which will update the UI.